src/objects.js

```diff
@@ -1,38 +1,71 @@
 // Complete the following underscore functions.
 // Reference http://underscorejs.org/ for examples.
+const arrayMethods = require('../src/arrays');
 
 const keys = (obj) => {
   // Retrieve all the names of the object's properties.
   // Return the keys as strings in an array.
   // Based on http://underscorejs.org/#keys
+
+  // not sure how to do this without just using an Object method
+  return arrayMethods.reduce(Object.entries(obj), (array, [key, value]) => {
```

Comment:
functional and shows you understand the concepts of the methods you've created already! Nice work!